src/components/backup/BackupConfirmPasswordStep.js

```diff
@@ -0,0 +1,251 @@
+  padding-left: 19;
+  padding-right: 40;
+  padding-top: 15;
+`;
+
+const IconWrapper = styled(View)`
+  margin-bottom: 12;
+  height: 22;
+  position: absolute;
+  right: 12;
+  top: 12;
+  width: 22;
+`;
+
+const Title = styled(Text).attrs({
+  size: 'big',
+  weight: 'bold',
+})`
+  margin-bottom: 12;
+`;
+
+const DescriptionText = styled(Text).attrs({
+  align: 'center',
+  color: colors.alpha(colors.blueGreyDark, 0.5),
+  lineHeight: 'looser',
+  size: 'large',
+})`
+  padding-bottom: 30;
+  padding-left: 50;
+  padding-right: 50;
+`;
+
+const WarningIcon = () => (
+  <IconWrapper>
+    <Icon color={colors.orangeLight} name="warningCircled" size={22} />
+  </IconWrapper>
+);
+
+const TopIcon = () => (
+  <GradientText
+    align="center"
+    angle={false}
+    colors={['#FFB114', '#FF54BB', '#00F0FF']}
+    end={{ x: 0, y: 0 }}
+    letterSpacing="roundedTight"
+    size={52}
+    start={{ x: 1, y: 1 }}
+    steps={[0, 0.5, 1]}
+    weight="bold"
+  >
+    <Text size={52}>􀙶</Text>
+  </GradientText>
+);
+
+const BackupConfirmPasswordStep = () => {
+  const { params } = useRoute();
+  const dispatch = useDispatch();
+  const walletCloudBackup = useWalletCloudBackup();
+  const [validPassword, setValidPassword] = useState(false);
+  const [passwordFocused, setPasswordFocused] = useState(true);
+  const [password, setPassword] = useState('');
+  const [label, setLabel] = useState('􀎽 Confirm Backup');
+  const passwordRef = useRef();
+  const { latestBackup, selectedWallet } = useWallets();
+
+  useEffect(() => {
+    const fetchPasswordIfPossible = async () => {
+      const pwd = await fetchBackupPassword();
+      if (pwd) {
+        setPassword(pwd);
+      }
+    };
+
+    fetchPasswordIfPossible();
+  }, []);
+
+  const onPasswordFocus = useCallback(() => {
+    setPasswordFocused(true);
+  }, []);
+
+  const onPasswordBlur = useCallback(() => {
+    setPasswordFocused(false);
+  }, []);
+
+  useEffect(() => {
+    let passwordIsValid = false;
+
+    if (isCloudBackupPasswordValid(password)) {
+      passwordIsValid = true;
+      setLabel(`􀑙 Add to iCloud Backup`);
+    }
+    setValidPassword(passwordIsValid);
+  }, [password, passwordFocused]);
+
+  const onPasswordChange = useCallback(
+    ({ nativeEvent: { text: inputText } }) => {
+      setPassword(inputText);
+    },
+    []
+  );
+
+  const onSubmit = useCallback(async () => {
+    const walletId = params?.walletId || selectedWallet.id;
+    await walletCloudBackup({
+      latestBackup,
+      onError: error => {
+        logger.sentry('Error while calling walletCloudBackup');
+        error && captureException(error);
+        passwordRef.current?.focus();
+        dispatch(setIsWalletLoading(null));
+        Alert.alert('Error while trying to backup');
+      },
+      password,
+      walletId,
+    });
+    setTimeout(() => {
+      Alert.alert('Your wallet has been backed up succesfully!');
+    }, 1000);
+  }, [
+    dispatch,
+    latestBackup,
+    params.walletId,
```

Comment:
I swear we were able to do params?.walletId without the linter complaining (at least, I know it's supported in some later version of one of the react linters, but perhaps not picked up in satya config yet). Anyways, if params doesn't exist, it'll still throw here right, so for now either we should pass in params or set params?.walletId outside to some const and pass in that const (gross I know)

Reply:
Resolved in: https://github.com/rainbow-me/rainbow/pull/1003